### ISSUE TITLE
Top level results get filtered out if nested connections with interfaces are empty when sorting

### ISSUE THREAD

**Issue by angrykoala** [neo4j/graphql#4532]
Top level results get filtered if a query is the connections to an interface are empty and these are sorted.

This happens on GraphQL 4.x
The following instructions to reproduce are for the coming up GraphQL 5.0.0 version

Given the following schema:

```graphql
type Inventory {
    id: ID
    children: [Scenario!]!
        @relationship(type: "HasChildren", properties: "InventoryChildRelation", direction: OUT)
}

interface Scenario {
    id: ID
}

type Image implements Scenario {
    id: ID
}

type Video implements Scenario {
    id: ID
}

type InventoryChildRelation @relationshipProperties {
    order: Int
}
```

And the following data:

```cypher
CREATE(i:Inventory {id: "i1"})
CREATE(i)-[:HasChildren { order: 3 }]->(c1:Image { id: "c3"})
CREATE(i)-[:HasChildren { order: 1 }]->(c2:Video { id: "c1"})
CREATE(i)-[:HasChildren { order: 2 }]->(c3:Video { id: "c2"})

CREATE(:Inventory {id: "i2"})
```

The following query:

```graphql
query {
  inventories {
      id
      childrenConnection(sort: { edge: { order: ASC } }) {
          edges {
              properties {
                  order
              }
              node {
                  id
              }
          }
      }
  }
}
```

Should return both `Inventory` items, however, the second one gets filtered out. This does not happen with connections to normal types or if sort is removed

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @angrykoala. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by angrykoala** [comment]
This was originally reported on [Discord](https://discord.com/channels/787399249741479977/818578492723036210/threads/1197229562245947513)

**Comment by angrykoala** [comment]
This fix will be available on 5.0.0
